static/build/.tmp_update/script/network/update_networking.sh

```diff
@@ -163,16 +163,18 @@ check_smbdstate() {
     fi
 }
 
+
 smbd_authed() {
     if flag_enabled smbdState; then
         if flag_enabled authsmbdState; then
             sed -i -e '/guest only/s/1/0/g' -e '/valid users = onion/s/^#//g' $sysdir/config/smb.conf
         else
             sed -i -e '/guest only/s/0/1/g' -e '/valid users = onion/ s/^#*/#/' $sysdir/config/smb.conf
         fi
-        if ! is_running smbd; then
-            $netscript/start_smbd.sh $PASS &
+        if is_running smbd; then
+            killall -9 smbd
```

Comment:
I think this needs to be more specific - the current logic will restart smbd on *ANY* network setting change. For example when toggling on/off ssh, I wouldn't want my samba transfer interupted.

Reply:
I might be wrong in my understanding, but isn't this how it works?

The smb auth toggle has the action `network_setSmbdAuthState`
which calls 
`network_setState(&network_state.auth_smbd, ".authsmbdState", ((ListItem *)pt)->value);
network_execServiceAuth("smbd");`

`set_state` just sets the config flag
`network_execServiceAuth("smbd");` runs `update_networking.sh smbd authed`

so the code you quoted (`smbd_authed`) should only be called when toggling smb authentication, from my understanding

Reply:
`update_networking` is called when you exit Tweaks - and all services are checked

Reply:
yes - `check_smbdstate` is run, which doesn't kill smbd unless it has been disabled and is still running.
but `smbd_authed` shouldn't be run at all unless changing the toggle?

Reply:
My bad... you're right, this is fine! 😅